src/Type/UnionTypeHelper.php

```diff
@@ -32,6 +34,24 @@ public static function getReferencedClasses(array $types): array
 		);
 	}
 
+	/**
+	 * @param Type[] $types
+	 * @return list<string>
+	 */
+	public static function getObjectClassNames(array $types): array
+	{
+		return array_values(
+			array_unique(
+				array_merge(
+					...array_map(
```

Comment:
I really don't like this notation, a good old foreach would be much better. I have no idea what's the result of array_merge with `...array_map` inside :)

As a bonus, with foreach, I can easily put some debugging statements between the calls.